### PR TITLE
Make cmake default config only an alias and cleanup param scope processing

### PR DESCRIPTION
**Pull request by jgoppert**
This improves the organization of the cmake configs for posix_sitl and nuttx_px4fmu-v2  so that the default config now is just an alias and all common files are in _common.cmake.

I needed this so when testing me PR I can switch default easily to IEKF from EKF2 so I can test my builds work without modifying the ekf2 config which was hard for me to maintain.

### CODE REVIEW

**Comment by jgoppert** [comment]
Odd, just reordering the modules due to the include layout caused a change in linking:

```bash
/opt/gcc/bin/../lib/gcc/arm-none-eabi/5.4.1/../../../../arm-none-eabi/bin/ld: src/firmware/nuttx/firmware_nuttx section `.data' will not fit in region `flash'
/opt/gcc/bin/../lib/gcc/arm-none-eabi/5.4.1/../../../../arm-none-eabi/bin/ld: region `flash' overflowed by 2836 bytes
```

I fixed it  by inserting estimator modules in the same spot they were.

**Comment by jgoppert** [comment]
I think this must be a bug on master? I haven't seen this before.

```bash
src/modules/param/px4_parameters.c:5098:12: fatal error: implicit conversion from 'long' to 'int32_t' (aka 'int') changes value from 3890735079 to -404232217 [-Wconstant-conversion]
                .val.i = 3890735079
                    ~    ^~~~~~~~~~
```

**Comment by bkueng** [comment]
One thing to consider: cmake param scoping does not work when using `include(configs/...)`.

compiling fails because of https://github.com/PX4/Firmware/blob/master/src/modules/syslink/syslink_params.c#L76: the number is larger than a max signed int, but still less than max unsigned int, thus still fitting but causing a conversion to a negative value.

**Comment by davids5** [comment]
@jgoppert - I expect the overflow is as @bkueng suggests, the parameters are not culled. (I lost track of how it is done now and why it was changed)

To see the differences. you can **Temporarily**  increase the flash size [here](https://github.com/PX4/Firmware/blob/master/nuttx-configs/px4fmu-v2/scripts/ld.script#L53-L54)  in your build. Then have a go as comparing the output of nm from master and your branch.

**Comment by jgoppert** [comment]
@bkueng Please review. This uses cmake to parse the cmake file and do param scoping instead of rolling our own cmake parser.

File: cmake/configs/nuttx_px4fmu-v2_common.cmake [hunk 1]
```diff
@@ -187,6 +181,12 @@ set(config_module_list
 	#examples/hwtest
 )
 
+set(config_module_list
+	${general_modules_section1}
+	${estimation_modules}
+	${general_modules_section2}
+)
+
```
Comment (jgoppert) [inline cmake/configs/nuttx_px4fmu-v2_common.cmake]: Without this I was still getting increased mem usage even after parm change, so I think it is best to leave it as is.

**Comment by jgoppert** [comment]
Rebased and CI is passing, I think this is ready for review and merge.

**Comment by davids5** [comment]
@mcharleb  @julianoes 

@jgoppert removed the ALL option in this PR. ALL was used to solve the issue https://github.com/PX4/Firmware/issues/4434 

Would one of you you please test this PR?

**Comment by jgoppert** [comment]
I've test flown with this now and it works well for my IEKF branch params.

**Comment by jgoppert** [comment]
@mcharleb  can you take a look. It would be good to get this in before we get another config update and I have to resolve the conflict. The reason I did this patch was because using the cmake include for configs wasn't working as I originally designed after the param scoping and I also suspect  that was your motivation for adding the scope all work around.

**Comment by dagar** [comment]
@jgoppert - that earlier CI failure was actually just circleci failing to fetch the docker image. Sorry for the late response.

**Comment by jgoppert** [comment]
@LorenzMeier can you try this out. It requires jinja2 and @priseborough was having trouble installing jinja2 on osx.I can update the wiki if we figure out the correct instructions.

**Comment by jgoppert** [comment]
Now that jinja is added for ci images and on the wiki, I think this is ready to merge. I don't anticipate any problems for @mcharleb. If I rebase again can I get it merged?

**Comment by jgoppert** [comment]
This is too hard to maintain and it is blocking me without patching it on my iekf branch due to error with param parsing, so I'm going to rebase on master and merge.  Please let me know if that is a problem @LorenzMeier @mcharleb @davids5.  I will watch my emails and quickly support any issues that crop up if I missed something but I don't expect any.

**Comment by LorenzMeier** [comment]
I can't really judge the potential impact, but I do care about you being able to continue and unblocked.

**Comment by LorenzMeier** [comment]
Meh, but its pretty super high risk and it will certainly kill all existing dev setups. I think some people will need to go through the adoption pain locally before we can merge this on master.

**Comment by jgoppert** [comment]
OK, will revert the name change that is causing all the maintenance problems, that is probably the best approach, then will add a check for jinja2 into cmake with a warning as suggested.

**Comment by LorenzMeier** [comment]
I'm happy to be Mac OS guinea pig.

**Comment by dagar** [comment]
I can help with testing/verification, although can we find a better solution for posit sitl? Any reason we can't have a single posit sitl config and select the estimator via init script choice?

**Comment by jgoppert** [comment]
Got the python module check working:

```bash
CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  couldn't find python module jinja2:

  		

  for debian systems try: 		

  	sudo apt-get install python-jinja2 		

  or for all other OSs/debian: 		

  	pip install jinja2

   (missing:  PY_JINJA2) 
```

@dagar It is true that we are currently building all of the estimators for posix, so why not. It would just require some makefile changes. I'll leave all of that alone for now, so no change from master to worry about.

**Comment by jgoppert** [comment]
@dagar probably testing on your end now, try removing jinja2, check the failure message, then install it, and make sure it builds again.

**Comment by dagar** [comment]
Looks good to me. The only difference I could find in px4fmu-v2 is that the mTecs params no longer end up in the generated source, which is great.


	CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
	  couldn't find python module jinja2:

	  

	  for debian systems try: 

		sudo apt-get install python-jinja2 

	  or for all other OSs/debian: 

		pip install jinja2

**Comment by jgoppert** [comment]
@LorenzMeier I think this is worth another look now.

**Comment by jgoppert** [comment]
@bkueng @LorenzMeier worth getting this in now?

**Comment by bkueng** [comment]
LGTM

**Pull request merged**